src/main/scala/firrtl/Emitter.scala

```diff
@@ -193,6 +193,15 @@ object EmitAllModulesAnnotation extends HasShellOptions {
       helpText = "Run the specified module emitter (one protobuf per module)",
       shortOption = Some("p"),
       helpValueName = Some("<chirrtl|mhigh|high|middle|low|low-opt>")
+    ),
+    new ShellOption[Unit](
+      longOption = "no-random-init",
+      toAnnotationSeq = _ =>
+        Seq(
+          CustomDefaultMemoryEmission(MemoryNoInit),
+          CustomDefaultRegisterEmission(useInitAsPreset = true, disableRandomization = true)
+        ),
+      helpText = "Disable random initialization for memory and registers"
```

Comment:
Combining these options together seems fine.  For now if users want a finer-grained API, they can use the annotations directly.

That said, we could consider adding separate options for `--no-random-init-mem` and `--no-random-init-reg`. I expect this is not the common case, though.  Alternatively, the CIRCT API for something like this is to add a `--lowering-options` argument that then takes tons of options.  We could do something like that here and have:

```
--emission-options=disableMemRandomization,disableRegisterRandomization
```